yumi:
  hiqp_joint_velocity_controller:
    #####      JOINT LIMITATIONS    #####
     hiqp_preload_jnt_limits:
      - link_frame: yumi_link_1_r
        limitations: [1.4835, -2.94, 2.94] #dq_max, lower joint limit, upper joint limit
      - link_frame: yumi_link_2_r
        limitations: [1.4835, -2.5, 0.75] 
      - link_frame: yumi_link_3_r
        limitations: [1.7453, -2.94, 2.94]
      - link_frame: yumi_link_4_r
        limitations: [1.3090, -2.16, 1.4]
      - link_frame: yumi_link_5_r
        limitations: [2.2689, -5.07, 5.07] 
      - link_frame: yumi_link_6_r
        limitations: [2.2689, -1.54, 2.41]
      - link_frame: yumi_link_7_r
        limitations: [2.2689, -4.0, 4.0] 

     hiqp_preload_geometric_primitives:
    #   #####      OBSTACLE AVOIDANCE PRIMITIVES    #####
      - name: floor_avoidance_plane
        type: plane 
        frame_id: yumi_body
        visible: true
        color: [0.5, 0.5, 0.5, 1.0]
        parameters: [0.0, 0.0, 1.0, 0.02]
      - name: yumi_link_5_r_avoidance_sphere
        type: sphere 
        frame_id: yumi_link_5_r
        visible: true
        color: [1.0, 0.0, 0.0, 0.6]
        parameters: [-0.023, -0.04, 0.1, 0.04]
      - name: yumi_link_6_r_avoidance_sphere
        type: sphere 
        frame_id: yumi_link_6_r
        visible: true
        color: [1.0, 0.0, 0.0, 0.6]
        parameters: [0.01, -0.01, -0.01, 0.06]
      - name: yumi_link_5_l_avoidance_sphere
        type: sphere 
        frame_id: yumi_link_5_l
        visible: true
        color: [1.0, 0.0, 0.0, 0.6]
        parameters: [-0.023, -0.04, 0.1, 0.04]
      - name: yumi_link_6_l_avoidance_sphere
        type: sphere 
        frame_id: yumi_link_6_l
        visible: true
        color: [1.0, 0.0, 0.0, 0.6]
        parameters: [0.01, -0.01, -0.01, 0.06]
      - name: gripper_r_base_avoidance_point_1
        type: point 
        frame_id: gripper_r_base
        visible: true
        color: [1.0, 0.0, 0.0, 0.6]
        parameters: [0.036, 0.035, 0.135]
      - name: gripper_r_base_avoidance_point_2
        type: point 
        frame_id: gripper_r_base
        visible: true
        color: [1.0, 0.0, 0.0, 0.6]
        parameters: [-0.036, 0.035, 0.135]
      - name: gripper_r_base_avoidance_point_3
        type: point 
        frame_id: gripper_r_base
        visible: true
        color: [1.0, 0.0, 0.0, 0.6]
        parameters: [0.036, -0.035, 0.135]
      - name: gripper_r_base_avoidance_point_4
        type: point 
        frame_id: gripper_r_base
        visible: true
        color: [1.0, 0.0, 0.0, 0.6]
        parameters: [-0.036, -0.035, 0.135]
      - name: gripper_l_base_avoidance_point_1
        type: point 
        frame_id: gripper_l_base
        visible: true
        color: [1.0, 0.0, 0.0, 0.6]
        parameters: [0.036, 0.035, 0.135]
      - name: gripper_l_base_avoidance_point_2
        type: point 
        frame_id: gripper_l_base
        visible: true
        color: [1.0, 0.0, 0.0, 0.6]
        parameters: [-0.036, 0.035, 0.135]
      - name: gripper_l_base_avoidance_point_3
        type: point 
        frame_id: gripper_l_base
        visible: true
        color: [1.0, 0.0, 0.0, 0.6]
        parameters: [0.036, -0.035, 0.135]
      - name: gripper_l_base_avoidance_point_4
        type: point 
        frame_id: gripper_l_base
        visible: true
        color: [1.0, 0.0, 0.0, 0.6]
        parameters: [-0.036, -0.035, 0.135]

    #   #####      OTHER PRIMITIVES    #####
    #   # - name: experiment_gripper_point
    #   #   type: point 
    #   #   frame_id: gripper_r_base
    #   #   visible: true
    #   #   color: [0.0, 0.0, 1.0, 0.6]
    #   #   parameters: [0.0, 0.0, 0.1]
    #   # - name: experiment_gripper_line_z
    #   #   type: line 
    #   #   frame_id: gripper_r_base
    #   #   visible: true
    #   #   color: [0.0, 0.0, 1.0, 0.6]
    #   #   parameters: [0.0, 0.0, 1.0, 0.0, 0.0, 0.1]
    #   # - name: experiment_gripper_line_y
    #   #   type: line 
    #   #   frame_id: gripper_r_base
    #   #   visible: true
    #   #   color: [1.0, 0.0, 1.0, 0.6]
    #   #   parameters: [0.0, 1.0, 0.0, 0.0, 0.0, 0.0]
    #   # - name: experiment_plane
    #   #   type: plane 
    #   #   frame_id: world
    #   #   visible: true
    #   #   color: [1.0, 1.0, 1.0, 0.6]
    #   #   parameters: [0.0, 0.0, 1.0, 0.215]
    #   # - name: experiment_cylinder
    #   #   type: cylinder
    #   #   frame_id: world
    #   #   visible: true
    #   #   color: [0.0, 1.0, 0.0, 0.6]
    #   #   parameters: [0.0, 0.0, 1.0, 0.5, 0.0, 0.115, 0.1, 0.1]

     hiqp_preload_tasks:
      #####      OBSTACLE AVOIDANCE TASKS    #####
      # - name: yumi_link_5_r_l_avoidance
      #   priority: 2
      #   visible: true
      #   active: true
      #   def_params: ['TDefGeomProj', 'sphere', 'sphere', 'yumi_link_5_l_avoidance_sphere > yumi_link_5_r_avoidance_sphere']
      #   dyn_params: ['TDynFirstOrder', '10.0']
      # - name: yumi_link_6_r_l_avoidance
      #   priority: 2
      #   visible: true
      #   active: true
      #   def_params: ['TDefGeomProj', 'sphere', 'sphere', 'yumi_link_6_l_avoidance_sphere > yumi_link_6_r_avoidance_sphere']
      #   dyn_params: ['TDynFirstOrder', '10.0']
      # - name: yumi_link_5_l_6_r_avoidance
      #   priority: 2
      #   visible: true
      #   active: true
      #   def_params: ['TDefGeomProj', 'sphere', 'sphere', 'yumi_link_5_l_avoidance_sphere > yumi_link_6_r_avoidance_sphere']
      #   dyn_params: ['TDynFirstOrder', '10.0']
      # - name: yumi_link_5_r_6_l_avoidance
      #   priority: 2
      #   visible: true
      #   active: true
      #   def_params: ['TDefGeomProj', 'sphere', 'sphere', 'yumi_link_6_l_avoidance_sphere > yumi_link_5_r_avoidance_sphere']
      #   dyn_params: ['TDynFirstOrder', '10.0']
      - name: yumi_link_5_l_floor_avoidance
        priority: 2
        visible: true
        active: true
        def_params: ['TDefGeomProj', 'sphere', 'plane', 'yumi_link_5_l_avoidance_sphere > floor_avoidance_plane']
        dyn_params: ['TDynFirstOrder', '10.0']
      - name: yumi_link_6_l_floor_avoidance
        priority: 2
        visible: true
        active: true
        def_params: ['TDefGeomProj', 'sphere', 'plane', 'yumi_link_6_l_avoidance_sphere > floor_avoidance_plane']
        dyn_params: ['TDynFirstOrder', '10.0']
      - name: yumi_link_5_r_floor_avoidance
        priority: 2
        visible: true
        active: true
        def_params: ['TDefGeomProj', 'sphere', 'plane', 'yumi_link_5_r_avoidance_sphere > floor_avoidance_plane']
        dyn_params: ['TDynFirstOrder', '10.0']
      - name: yumi_link_6_r_floor_avoidance
        priority: 2
        visible: true
        active: true
        def_params: ['TDefGeomProj', 'sphere', 'plane', 'yumi_link_6_r_avoidance_sphere > floor_avoidance_plane']
        dyn_params: ['TDynFirstOrder', '10.0']
      - name: yumi_r_gripper_floor_avoidance_1
        priority: 2
        visible: true
        active: true
        def_params: ['TDefGeomProj', 'point', 'plane', 'gripper_r_base_avoidance_point_1 > floor_avoidance_plane']
        dyn_params: ['TDynFirstOrder', '10.0']
      - name: yumi_r_gripper_floor_avoidance_2
        priority: 2
        visible: true
        active: true
        def_params: ['TDefGeomProj', 'point', 'plane', 'gripper_r_base_avoidance_point_2 > floor_avoidance_plane']
        dyn_params: ['TDynFirstOrder', '10.0']
      - name: yumi_r_gripper_floor_avoidance_3
        priority: 2
        visible: true
        active: true
        def_params: ['TDefGeomProj', 'point', 'plane', 'gripper_r_base_avoidance_point_3 > floor_avoidance_plane']
        dyn_params: ['TDynFirstOrder', '10.0']
      - name: yumi_r_gripper_floor_avoidance_4
        priority: 2
        visible: true
        active: true
        def_params: ['TDefGeomProj', 'point', 'plane', 'gripper_r_base_avoidance_point_4 > floor_avoidance_plane']
        dyn_params: ['TDynFirstOrder', '10.0']
      - name: yumi_l_gripper_floor_avoidance_1
        priority: 2
        visible: true
        active: true
        def_params: ['TDefGeomProj', 'point', 'plane', 'gripper_l_base_avoidance_point_1 > floor_avoidance_plane']
        dyn_params: ['TDynFirstOrder', '10.0']
      - name: yumi_l_gripper_floor_avoidance_2
        priority: 2
        visible: true
        active: true
        def_params: ['TDefGeomProj', 'point', 'plane', 'gripper_l_base_avoidance_point_2 > floor_avoidance_plane']
        dyn_params: ['TDynFirstOrder', '10.0']
      - name: yumi_l_gripper_floor_avoidance_3
        priority: 2
        visible: true
        active: true
        def_params: ['TDefGeomProj', 'point', 'plane', 'gripper_l_base_avoidance_point_3 > floor_avoidance_plane']
        dyn_params: ['TDynFirstOrder', '10.0']
      - name: yumi_l_gripper_floor_avoidance_4
        priority: 2
        visible: true
        active: true
        def_params: ['TDefGeomProj', 'point', 'plane', 'gripper_l_base_avoidance_point_4 > floor_avoidance_plane']
        dyn_params: ['TDynFirstOrder', '10.0']

    #####      OTHER TASKS    #####
